src/XrdTls/XrdTlsTempCA.cc

```diff
@@ -202,10 +212,17 @@ CRLSet::processFile(file_smart_ptr &fp, const std::string &fname)
         //m_log.Emsg("CRLset", "New CRL with hash", fname.c_str(), hash_ptr);
         m_known_crls.insert(hash_ptr);
 
-        if (!xrd_crl->ToFile(outputfp.get())) {
+        if(xrd_crl->hasCriticalExtension()) {
+          // Issue https://github.com/xrootd/xrootd/issues/2065
+          // This CRL will be put at the end of the bundled file
+          m_crls_critical_extension.emplace_back(std::move(xrd_crl));
+        } else {
+          // No critical extension found on that CRL, just insert it on the CRL bundled file
+          if (!xrd_crl->ToFile(outputfp.get())) {
             m_log.Emsg("CRLset", "Failed to write out CRL", fname.c_str());
             fflush(outputfp.get());
             return false;
+          }
         }
```

Comment:
It seems to me that you could declare the vector `m_crls_critical_extension` before the `for` loop, then process it here, no? You can then pass a reference to it to `processCRLWithCriticalExt` and avoid adding a member to the class.

Reply:
Yes, I did not want to change the signature of the member function, but sure, if you prefer it that way, it can be done :)

Reply:
Yes, I realize now that my suggestion doesn't make much sense as the CRLs with extensions would be processed at the end of each file, which doesn't help much. Let's leave it like this and wait for Andy's comments. Then I will merge this into devel for 5.6.2.